test/ui/BreadcrumbValuesListTest.ts

```diff
@@ -0,0 +1,113 @@
+import * as Mock from '../MockEnvironment';
+import { Facet } from '../../src/ui/Facet/Facet';
+import { FacetValue } from '../../src/ui/Facet/FacetValues';
+import { BreadcrumbValueElement } from '../../src/ui/Facet/BreadcrumbValueElement';
+import * as _ from 'underscore';
+import { $$ } from '../Test';
+import { FakeResults } from '../Fake';
+import { IIndexFieldValue } from '../../src/rest/FieldValue';
+
+export function BreadcrumbValueListTest() {
+  describe('BreadcrumbValuesListTest', () => {
+    let facet: Facet;
+    let facetValues: FacetValue[];
+
+    beforeEach(() => {
+      facet = Mock.mock(Facet);
+      facet.options = {};
+      facet.options.numberOfValuesInBreadcrumb = 5;
+      facet.getValueCaption = jasmine.createSpy('getValueCaption').and.callFake((facetValue: IIndexFieldValue) => facetValue.value);
+      facetValues = _.map(FakeResults.createFakeFieldValues('foo', 3), fieldValue => {
+        const created = FacetValue.createFromFieldValue(fieldValue);
+        created.selected = true;
+        return created;
+      });
+    });
+
+    it('should support rendering as a simple string and not an HTML element', () => {
+      facet.options.title = 'My facet title';
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const builtAsString = breadcrumbList.buildAsString();
+      expect(_.isString(builtAsString)).toBe(true);
+      expect(builtAsString).toContain('My facet title');
+      expect(builtAsString).toContain('foo0');
+      expect(builtAsString).toContain('foo1');
+      expect(builtAsString).toContain('foo2');
+    });
+
+    it('should output the title with the facet title option', () => {
+      facet.options.title = 'My facet title';
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const title = $$(built).find('.coveo-facet-breadcrumb-title');
+      expect($$(title).text()).toContain('My facet title');
+    });
+
+    it('should display a clickable label for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const values = $$(built).findAll('.coveo-facet-breadcrumb-value');
+      expect(values.length).toBe(3);
+    });
+
+    it('should display a caption for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const captions = $$(built).findAll('.coveo-facet-breadcrumb-caption');
+      expect($$(captions[0]).text()).toContain('foo0');
+      expect($$(captions[1]).text()).toContain('foo1');
+      expect($$(captions[2]).text()).toContain('foo2');
+    });
+
+    it('should display a clear button for each selected values', () => {
```

Comment:
Based of this test, is it possible to add another test that the clear button removes a breadcrumb from the value list?

Reply:
Like above, I think it would be better to test this properly in the `BreadcrumbValueElement` class instead.